src/components/signup.js

```diff
@@ -13,28 +27,97 @@ import VisibilityOff from '@mui/icons-material/VisibilityOff';
 //State Management - Context API
 import { UserContext } from '../contextAPI';
 
-
 export default function SignUp(props) {
    const userContext = useContext(UserContext)
+   const user = userContext.user
 
-   const { changeSignup,
-      valuesSignup,
-      signupErrors,
-      submitNewUser,
-      handleShowPass } = props
+   const { setAccountValues,
+      handleShowPass,
+      setUser,
+      signupFormValues,
+      setSignupFormValues,
+      navigate } = props
+
+   const [signupErrors, setSignupErrors] = useState(initialSignupValues)
+   const [disabled, setDisabled] = useState(initialDisabled)
 
    const onChangeSignup = event => {
       const { name, value, checked, type } = event.target
 
       const valueUsed = type === 'checkbox' ? checked : value;
 
-      changeSignup(name, valueUsed)
+      changeInputSignup(name, valueUsed)
    }
 
    const handleMouseDownPass = (event) => {
       event.preventDefault()
    }
 
+   //Validation Errors + changing input for Sign Up Page:
+   const changeInputSignup = (name, value) => {
+      yup
+         .reach(formSchemaSignup, name)
+         .validate(value)
+         .then(() => {
+            setSignupErrors({ ...signupErrors, [name]: '' })
+         })
+         .catch(err => {
+            setSignupErrors({ ...signupErrors, [name]: err.errors })
+         })
+
+      setSignupFormValues({ ...signupFormValues, [name]: value })
+   }
+
+   useEffect(() => {
+      formSchemaSignup.isValid(signupFormValues).then(validate => {
+         setDisabled(!validate)
+      })
+   }, [signupFormValues])
+
+   //Posting a new user to the user api when Signing Up
+   const registerNewUser = (newUser) => {
+      axiosWithAuth()
+         .post('auth/register', newUser)
+         .then(res => {
+            setUser(res.data.user)
+            setAccountValues(res.data.user)
```

Comment:
Could you have just one setState? Since you're setting the same data for both User and Account could you use either User or Account in wherever you need that data?